bool prepare_rotation
bool prepare_translation
---
bool prepared
